### ISSUE TITLE
DNS leak when upgrading ctrld

### ISSUE THREAD

**Issue by pmcarrion** [Control-D-Inc/ctrld#60]
There's a DNS leak when upgrading ctrld.

I just upgraded to the developer version using the quick install script:
`sudo sh -c 'sh -c "$(curl -sL https://api.controld.dev/dl)" -s RESOLVER_ID_HERE forced'`

As you can see below, the installer detects this is an upgrade, downloads the new binary, stops the service, replaces the old binary with the new one, makes it executable and restarts the service.

This is all expected, but if the installer detects this is an upgrade, there is no need to remove the DNS server `127.0.0.1` while upgrading ctrld. **This is when the DNS leak happens.**

The DNS leak also happens with the regular installer:
`sudo sh -c 'sh -c "$(curl -sL https://api.controld.com/dl)" -s RESOLVER_ID_HERE forced'`

Please modify the installer, so that the system's DNS server `127.0.0.1` is not touched when upgrading ctrld.

Furthermore, please specify the updated and previous version numbers of ctrld.

Finally, it would be nice to see the changelog of the developer version.

```
        __         .__       .___
  _____/  |________|  |    __| _/
_/ ___\   __\_  __ \  |   / __ |
\  \___|  |  |  | \/  |__/ /_/ |
 \___  >__|  |__|  |____/\____ |
     \/       installer       \/

---------------------
|    System Info    |
---------------------
OS Type      : darwin
OS Vendor    : MacOS
OS Version   : 13.4.1
Is Router    : no
Arch         : arm64
CPU          : Apple M1 Max
Free RAM     : 8015 MB / 65536 MB
---------------------
|  Install Details  |
---------------------
Resolver ID  : RESOLVER_ID_HERE
Binary URL   : https://assets.controld.dev/ctrld/darwin/arm64/ctrld
Install Path : /usr/local/bin
---------------------
 - Detected running process, this is an upgrade
 - Starting download
 - Stopping running process
 - Replacing old binary with the new one
 - Making binary executable
 - Launching /usr/local/bin/ctrld
---------------------
Jun 22 23:19:24.000 INF loading config file from: /etc/controld/ctrld.toml
Jun 22 23:19:24.000 INF fetching Controld D configuration from API: RESOLVER_ID_HERE mode=cd
Jun 22 23:19:25.000 INF generating ctrld config from Control-D configuration mode=cd
Jun 22 23:19:25.000 INF writing config file to: /etc/controld/ctrld.toml mode=cd
Jun 22 23:19:27.000 NTC Service started
```

**Comment by yegors** [comment]
Unsure how much of a problem this is. The cost of extra complexity to avoid this is not worth it, imo. "Leaking" DNS queries for a few seconds every time you upgrade seems rather trivial. 

Will keep this open for more feedback.

**Comment by pmcarrion** [comment]
When upgrading to the latest version of ctrld, the installer removed the DNS server (127.0.0.1) from all network instances (Wi-Fi and Ethernet), and set it up only on the active network interface (Ethernet) when the upgrade completed.

This caused a DNS leak When I disconnected Ethernet and roamed to Wi-Fi, as I didn’t notice the DNS server was removed from the Wi-Fi settings.

There is no need to remove the DNS server from network interfaces when upgrading ctrld.
